Replace LiveReload with Browsersync

JodiWarren opened the pull request:
The one caveat with this approach is that the user needs to add the local URL into their gulp file.

There's definitely plenty of room for improvement, but I thought a relatively quick and simple solution would at least improve everyone's experience immediately.

Closes #60

JodiWarren committed:
Replace LiveReload with Browsersync
Prevent reloading on admin, but allow admin-ajax to work.

Closes #60

timwright12 commented on gulpfile.babel.js:
```diff
@@ -1,18 +1,36 @@
-	gulp.watch( ['./assets/css/**/*.css', '!./assets/css/src/**/*.css'], gulp.series( 'cssprocess' ) );
-	gulp.watch( './assets/js/**/*.js', gulp.series( 'js' ) );
+	bs.init({
+		proxy: 'local-theme-url.test',
```
there's a search and replace value for URL in the generator, it's: tenup-scaffold.test - however, I think we need to find a way to bubble this up better rather than burying it in a gulp file. Part of the goal in creating the scaffolding is that it should work out of the box. I'd like to stay away from starting a pattern where we need to manually update lines of code before starting work.

JodiWarren replied:
Would you be more comfortable having either something like a `config.json` file, or maybe having the local URL in the `package.json` file?

colorful-tones replied:
I would vote for a `config.json` and even perhaps an optional `config-local.json` for Engineers to override certain things on their local's.

timwright12 commented on package.json:
```diff
@@ -44,6 +44,7 @@
     "babel-eslint": "^8.2.6",
     "babel-loader": "^7.1.5",
     "babel-preset-env": "^1.6.1",
+    "browser-sync": "^2.24.7",
```
Don't forget to remove gulp-livereload if we're not using it anymore. Also update the Readme file with a link to the new package being used

timwright12 reviewed (changes requested):
@JodiWarren Make sure the whole team signs off on this change over or has a chance to voice their opinion before any merge. Also make sure this change makes its way over to the plugin scaffold as well before merging

JodiWarren committed:
Add config files

- Add config.json and config-local.json for any theme-wide dev settings
- Remove gulp-livereload
- Update readme

JodiWarren commented:
I've now added a `config.json` file to hold the local url. I don't feel 100% comfortable with it at the moment, as it feels a little overcomplicated, but we need some structured way of providing the vhost URL to Browsersync.

A smoother option would be to request it as part of the installation process when using the `create-10up` CLI tool.

In theory, I think we could automatically figure it out it when using WP Local Docker, but I don't know to what degree of certainty.

timwright12 commented:
@JodiWarren I think storing the URL is a bigger conversation we need to figure out. If we're going to use a file, I'd prefer to store it in the package.json file similar to how the testing URLs are stored

timwright12 commented on README.md:
```diff
@@ -7,16 +7,19 @@ At 10up, we strive to provide digital products that yield a top-notch user exper
+2. [Gulp](https://gulpjs.com/) - Gulp is used as the main task runner, it runs PostCSS, processes images, handles SVG sprites (if needed), and executes Webpack.
+3. [Webpack](https://webpack.js.org/) - Webpack is used to process the JavaScript.
 4. [Composer](https://getcomposer.org/) - Composer is used to manage PHP.
+5. [Browsersync](https://browsersync.io/) - Browsersync is used for effortless previews of your site across your local devices.
```
@JodiWarren this goes down in the other list

JodiWarren replied:
@timwright12 I have it in the other list as well, but I thought it was worth making people aware of it, as it's a useful part of the toolkit.

timwright12 replied:
@JodiWarren this is more so the list for global stuff you need (like node) installed rather than secondary tools

JodiWarren replied:
@timwright12 Not to be overly pedantic, but should we remove Gulp and Webpack too, or maybe move them into a different section? They're not external dependencies either, and it might suggest that people need them installed globally to use the build system.

timwright12 replied:
We can deal with that in a future PR, for the purposes of this one, let's move Browsersync down to the other section.

timwright12 commented:
@JodiWarren  Is this PR replaced with https://github.com/10up/theme-scaffold/pull/66? If so, please close/delete

JodiWarren commented:
Replaced with #66
